src/ci/shared.sh

```diff
@@ -53,7 +53,8 @@ function isLinux {
 }
 
 function isKnownToBeMingwBuild {
-    isGitHubActions && [[ "${CI_JOB_NAME}" == *mingw ]]
+    # CI_JOB_NAME must contain "mingw" to be considered a MinGW build.
+    isGitHubActions && [[ "${CI_JOB_NAME}" == *mingw* ]]
```

Comment:
This is going to affect the job `mingw-check`, too. Is this fine?

https://github.com/rust-lang/rust/blob/251dc8ad84492c792a7600d8c5fef2ec868a36a7/src/ci/github-actions/jobs.yml#L221

Reply:
It does seem to affect some nontrivial behavior, so I would rather keep the original behavior for `mingw-check` in this PR. We can just hard-code it here.

Reply:
we could do:

- [ ] match on the trailing number (i.e. `-N`)

Suggested by JD 🙌

Reply:
Why not :) Anything works.

Reply:
done in https://github.com/rust-lang/rust/pull/132721/commits/0f176c85ccce7c64a5be1cc0990f574d9056cde5 👍